rspace/src/main/scala/coop/rchain/rspace/StoreEventsCounter.scala

```diff
@@ -0,0 +1,106 @@
+package coop.rchain.rspace
+import java.util.concurrent.atomic.{AtomicLong, AtomicReference}
+import scala.annotation.tailrec
+
+case class StoreCounters(sizeOnDisk: Long,
+                         dataEntries: Long,
+                         consumesCount: Long,
+                         consumeAvgMilliseconds: Double,
+                         producesCount: Long,
+                         produceAvgMilliseconds: Double)
+
+/**
+  * Conters rspace produce and consume calls.
+  * Returns counted number of cycles and avg times
+  * Uses BigIntegers to accumulate total times
+  * w/o rounding errors
+  *
+  * Lock-free. Non wait-free.
+  *
+  * For percentiles here is an algorithm that will not drain
+  * memory: http://www.cs.wustl.edu/~jain/papers/ftp/psqr.pdf,
+  */
+private[rspace] class StoreEventsCounter {
+  private[this] val producesCount   = new AtomicLong
+  private[this] val producesSumTime = new AtomicReference[BigInt](BigInt(0))
+
+  private[this] val consumesCount   = new AtomicLong
+  private[this] val consumesSumTime = new AtomicReference[BigInt](BigInt(0))
+
+  private[rspace] def reset(): Unit = {
+    producesCount.set(0)
+    producesSumTime.set(0)
+
+    consumesCount.set(0)
+    consumesSumTime.set(0)
+  }
+
+  /**
+    * @return a tuple (producesCount, consumesCount) just for unit-tests
+    */
+  private[rspace] def getCounts: (Long, Long) = (producesCount.get, consumesCount.get)
```

Comment:
Let's have separate `getProducesCount` and `getConsumesCount` methods.  Yes, it adds another line of code to the tests, but it's much easier to understand the tests in context if we know what we are getting by looking at the function call.

Reply:
Done.